indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/Prover.java

```diff
@@ -64,6 +65,99 @@ public void init(String masterSecretName) throws IndyException, ExecutionExcepti
         return Anoncreds.proverStoreClaimOffer(wallet.getWallet(), claimOffer.toJSON());
     }
 
+    /**
+     * Proves the proofRequest using the stored claims
+     *
+     * @param proofRequest
+     * @param attributes This map is used to get the correct claim, if multiple referents are present, or to provide self-attested attributes
+     * @return
+     */
+    public CompletableFuture<Proof> proofRequestToProof(ProofRequest proofRequest, Map<String, String> attributes) throws JsonProcessingException, IndyException {
+        return Anoncreds.proverGetClaimsForProofReq(wallet.getWallet(), proofRequest.toJSON())
+                .thenApply(wrapException(claimsForProofRequestJson -> {
+                    return JSONUtil.mapper.readValue(claimsForProofRequestJson, ClaimsForRequest.class);
+                }))
+                .thenCompose(wrapException(claimsForRequest -> createProofFromClaims(proofRequest, claimsForRequest, attributes, proofRequest.getTheirDid())));
+
+    }
+
+    CompletableFuture<Proof> createProofFromClaims(ProofRequest proofRequest, ClaimsForRequest claimsForRequest, Map<String, String> attributes, String theirDid) throws JsonProcessingException {
```

Comment:
This is going to be a nightmare to test. Please break down this monolith function into smaller ones.

Reply:
Refactored.